### DISCUSSION TITLE
TRIQS and Wannier90 - not all files are created

### DISCUSSION THREAD

**Discussion by AQuantumGuy** [TRIQS/triqs#854] [Q&A] [answered]
Hi everyone,
I am not sure if raising Wannier90 issue here is fine but since I'm trying to setup TRIQS code, I thought to write here.
I am trying to follow the [tutorial](https://github.com/wannier-developers/wannier-tutorials/blob/master/2022_05_Trieste/DAY3_AM_TRIQS) for TRIQS as  I want to try the framework Quantum Espresso + Wannier90 + TRIQS. The problem is, after I successfully run my calculations with QE and W90, I tried to setup the code for TRIQS. The problem is right at the beginning, when I try to compare QE dos with W90 dos, I cannot do that because W90 has not created the DOS file for me. In the W90 input file I have stated "dos = true" (like in the example), but it doesn't create any file for it.

I tried going beyond that and I found out that the software haven't created my .inp file as well. Also, I tried to compare my W90 bands with QE bands, and it only shows QE bands in the plot and not the W90 one. I do have both files, but when I try to simulate on jupyter notebook, W90 bands are not present.

Could someone tell me why this happens and how to fix it? Just to confirm, I do not get errors when I complete the W90 calculations for which I run three codes to "wannier90.x -pp FILENAME" then "pw2wannier90.x < FILENAME > FILENAME and then wannier90.x FILENAME. These are the only three codes I run the W90 simulation.

**Comment by phibeck** [comment]
Hi,

thanks for your question!

1. Regarding your first point, in order to compute the DOS in Wannier90 you have to use the post-processing program as `postw90.x seedname`, see the Wannier90 User guide http://www.wannier.org/support/ for more information.
2. The `seedname.inp` file is not created automatically, since this really depends on your specific interest. You have to create this file yourself as described here: https://triqs.github.io/dft_tools/latest/guide/conv_W90.html Let me know if you have questions about this, or perhaps open another discussion on TRIQS/dft_tools in case you run into trouble.
3. Comparing the bands of QE with those of Wannier90 - in principle this should work and it makes sense to figure this out before continuing. My first guess is a mismatch in the chemical potential, since Wannier90 knows nothing about that. Could it be that your Wannier90 bands are just shifted up in energy?

Best,
Sophie

**Reply by AQuantumGuy** [reply]
Hi,
Thank you very much for your reply. I understand the first 2 points and I successfully managed to obtain those files. As per the Wannier90 bands and QE, I am not sure why, but I only had 1 tight binding band (I specified 22 Wannier Bands for the input) and its location was at around -4000 eV which is so far from the Fermi level on which I'm focusing. 

Could you let me know what I might have done wrong?

For context I'm working with Titanium Dioxide (rutile phase) and I'm interested in certain bands between 2eV and 5eV as these contain Ti-d states.

Sincerely,

**Comment by phibeck** [comment]
Hi,

it is very difficult to guess from this little information what went wrong. Since this is not a TRIQS issue, I recommend that you post this question on the Wannier90 user mailing list. It should be self-contained, i.e. contain all necessary input files. Here's the link to the email forum: http://www.wannier.org/support/

Best,
Sophie

**Reply by AQuantumGuy** [reply]
Hey, 
I wasn't sure whether this was a Wannier90 issue because I can plot (using gnuplot) and I get my targeted Wannier bands which are very similar to my QE result. I asked here because when I run the following code (from the tutorial) on Jupyter Notebook, I get **H(R=0): -4000**

```
# get_dmft_bands extracts H(k) from the above specified path
tb_bands_data, alatt_k_w_path, freq_dict, dft_mu = pcb.get_dmft_bands(orbital_order_to = [0], **w90_dict, **tb_bands)
```
When I try to see that band, then I have one red line far from QE bands. 

Sincerely,

**Comment by phibeck** [comment, marked as answer]
Hi,
thanks for clarifying - this is a very different case then and indeed the right place to report (TRIQS/solid_dmft to be precise). Could you please share your definition of `w90_dict`? It sounds like the issue is located somewhere there.
Best,
Sophie

**Reply by AQuantumGuy** [reply]
Below is the one I've tried to run (similar as in tutorial)

```
# w90 information
w90_dict = {'w90_path': './QE/mlwf/',
        	'w90_seed': 'tio',
        	'add_spin': False, 'add_lambda': None,
        	'n_orb': 1,
        	'mu': fermi,
        	'orbital_order_w90': [0]}

# plotting options
plot_dict = {'colorscheme_bands': 'coolwarm', 'proj_on_orb': None, 'vmin': 0.0}

# tight-binding-like bands we want to plot from the given W90 Hamiltonian
tb_bands = {'bands_path': [('G', 'X'), ('X', 'M'), ('M', 'G'), ('G', 'Z'), ('Z', 'R'), ('R', 'A'), ('A', 'Z')], 'n_k': 50,
        	'G': [ 0.0, 0.0, 0.0], 'X': [0.5, 0.0, 0.0], 'M': [0.5, -0.5, 0.0], 'Z': [0.0, 0.0, 0.5], 'R': [0.5, 0.0, 0.5], 'A': [0.5, -0.5, 0.5]}

# get_dmft_bands extracts H(k) from the above specified path
tb_bands_data, alatt_k_w_path, freq_dict, dft_mu = pcb.get_dmft_bands(orbital_order_to = [0], **w90_dict, **tb_bands)

# also load the bands.dat.gnu file for a comparison
band_data = np.loadtxt('./QE/bands/tio_bands.dat.gnu')
```

**Reply by phibeck** [reply]
Try `n_orb = 22` instead of 1. You will probably have to define as many orbitals for the parameters `orbital_order_w90` and `orbital_order_to`.

**Reply by AQuantumGuy** [reply]
Thank you so much Sophie. It works! I wasn't aware that in n_orb I had to specify my targeted bands. Truley appreciated your prompt replies.

Sincerely,
